### PR TITLE
[GH-#85] Validation for difficulty target

### LINKED ISSUE #85: validate difficulty target
- [ ] validate if difficulty target in header matches the calculation rules

### PR DESCRIPTION
**Pull request by nedelcho-delchev-tues**
closes #85

### CODE REVIEW

File: apps/aecore/lib/aecore/miner/worker.ex [hunk 1]
```diff
@@ -138,7 +138,17 @@ defmodule Aecore.Miner.Worker do
...
     end
     try do
-      BlockValidation.validate_block!(latest_block, previous_block, chain_state)
+
+      BlockValidation.validate_block!(latest_block, previous_block, chain_state, blocks_for_difficulty_calculation)
+
+    catch
+      {:error, _message} ->
+        Logger.error(fn ->
+          "Failed to mine block"
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/miner/worker.ex]: can we log the message here as well?

**Comment by nedelcho-delchev-tues** [comment]
Not done yet, don't merge

**Comment by nedelcho-delchev-tues** [comment]
The issue is fixed, you can review it @meivantodorov, @cytadela8, @andonov.

File: apps/aecore/lib/aecore/miner/worker.ex [hunk 2]
```diff
@@ -127,21 +137,37 @@ defmodule Aecore.Miner.Worker do
...
     chain_state = Chain.chain_state(latest_block_hash)
 
-    txs_list = Map.values(Pool.get_pool())
-    ordered_txs_list = Enum.sort(txs_list, fn(tx1, tx2) -> tx1.data.nonce < tx2.data.nonce end)
+    blocks_for_difficulty_validation = if(latest_block.header.height == 0) do
+      Chain.get_blocks(latest_block_hash, Difficulty.get_number_of_blocks())
+    else
+      [_ | something] = Chain.get_blocks(latest_block_hash, Difficulty.get_number_of_blocks() + 1)
+      something
+    end
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/miner/worker.ex]: I trust this logic is correct. *it looks weird*

Comment (thepiwo) [reply apps/aecore/lib/aecore/miner/worker.ex]: @nedelcho-delchev-tues maybe we can optimize this somehow to be better readable and add some comment

Comment (nedelcho-delchev-tues) [reply apps/aecore/lib/aecore/miner/worker.ex]: I pushed a change directly to master, you can see if it is better and more understandable.
@cytadela8

**Pull request merged**